### PR TITLE
[libunifex] Update to 2021-12-07

### PR DESCRIPTION
**Pull request by zeyangl**
**Describe the pull request**

- #### What does your PR fix?  
Update libunifex to 2021-12-07

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
tested base and feature [test] compile:
  windows - verified
  linux - verified
  mac - verified

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by zeyangl** [comment]
@JonLiu1993 
This is a redo of https://github.com/microsoft/vcpkg/pull/21483
Updated libunifex version and verified that [test] feature works on mac/linux/windows triplets

File: ports/libunifex/portfile.cmake [hunk 1]
```diff
@@ -3,12 +3,13 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
...
-    REF e36b43834329acc75f99910316d3ecec15c0f665
-    SHA512 58f738b49d18982fd3916500c78e8090a266738ec4a8ba416b004fa2c7db718db5ce0e27f5e4e22eaddcc27c8d1618bcb2bc080eee1a31d98543e2595ccc1135
+    REF 9df21c58d34ce8a1cd3b15c3a7347495e29417a0
+    SHA512 2a1c6dbe0e66893be888c301e53b5d2813049047a817e9e1562acf2dec4ae79dec7db90409cd415c59238f2f3e2738e52f86a424041459390c720568ee52d860 
     HEAD_REF master
     PATCHES
-        fix-install.patch
-        allow-warnings.patch
+    #    fix-install.patch
+    #    allow-warnings.patch
```
Comment (JonLiu1993) [inline ports/libunifex/portfile.cmake:11]: @zeyangl ,Thanks for your pr, If these patches are no longer needed, please remove them.
```suggestion
```

File: versions/l-/libunifex.json [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "a6a3d0d1777eeec2ac0363c445d98e0e9ca43908",
+      "version-date": "2021-12-07",
+      "port-version": 0
+    },
+    {
+      "git-tree": "026508227fab71e346483fbfe3f349981f31de86",
+      "version-date": "2021-11-06",
+      "port-version": 0
+    },
```
Comment (JonLiu1993) [inline versions/l-/libunifex.json:12]:
```suggestion
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by zeyangl** [comment]
removed unused as requested

**Review by BillyONeal — APPROVED** [review]
I verified that there are now official targets https://github.com/facebookexperimental/libunifex/blob/21e72b742fa8df8935c606f500f99b4b458c9adb/source/CMakeLists.txt#L87

**Comment by zeyangl** [comment]
If there's anything else left to do please let me know.

File: ports/libunifex/fix-compile-error.patch [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
- if (MSVC)
-     # warning level 3 and all warnings as errors
--    add_compile_options(/W3 /WX)
-+    add_compile_options(/W3)
- else()
-     # lots of warnings and all warnings as errors
--    add_compile_options(-Wall -Wextra -pedantic -Werror)
-+    add_compile_options(-Wall -Wextra -pedantic)
- endif()
+diff --git a/cmake/unifex_env.cmake b/cmake/unifex_env.cmake
```
Comment (JonLiu1993) [inline ports/libunifex/fix-compile-error.patch:1]: @zeyangl ,Could you please keep the original format of EOL for this patch, maybe LF?

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/libunifex/fix-compile-error.patch (hunk 3 above)
Comment (zeyangl) [reply ports/libunifex/fix-compile-error.patch:1]: I'm a bit confused here. I double checked my local file and its EOL is LF.

**Comment by BillyONeal** [comment]
Thanks for the update!

**Comment by zeyangl** [comment]
hooray! my first ever pr to vcpkg.

**Pull request merged**